config/api/routes/lock.php

```diff
@@ -9,44 +11,84 @@
         'pattern' => '(:all)/lock',
         'method'  => 'GET',
         'action'  => function (string $path) {
-            return $this->parent($path)->lock()->get();
+            if ($lock = $this->parent($path)->lock()) {
+                $locked = $lock->get();
+            }
+
+            return [
+                'supported' => isset($locked) === true,
```

Comment:
Hm, while this works I think the code is not easy to understand if you don't know what's going on.

What about something like this (same for the `GET unlock` route):

```php
if ($lock = $this->parent($path)->lock()) {
    return [
        'supported' => true,
        'locked'    => $lock->get()
    ];
}

return [
    'supported' => false,
    'locked'    => null
]
```

Just a few more lines but a clear separation between the two cases.

Reply:
I agree with @lukasbestle